exchanges/btcmarkets/btcmarkets_websocket.go

```diff
@@ -205,3 +228,27 @@ func (b *BTCMarkets) Subscribe(channelToSubscribe wshandler.WebsocketChannelSubs
 	}
 	return b.WebsocketConn.SendMessage(req)
 }
+
+// Login logs in allowing private ws events
+func (b *BTCMarkets) generateAuthSubscriptions() error {
+	var authSub WsAuthSubscribe
+	signTime := strconv.FormatInt(time.Now().UTC().UnixNano()/1000000, 10)
+	strToSign := "/users/self/subscribe" + "\n" + signTime
+	tempSign := crypto.GetHMAC(crypto.HashSHA512,
+		[]byte(strToSign),
+		[]byte(b.API.Credentials.Secret))
+	sign := crypto.Base64Encode(tempSign)
+	markets, err := b.GetMarkets()
+	if err != nil {
+		return err
+	}
+	for x := range markets {
+		authSub.MarketIDs = append(authSub.MarketIDs, markets[x].MarketID)
+	}
+	authSub.Channels = []string{"fundChange", "heartbeat", "orderChange"}
+	authSub.Key = b.API.Credentials.Key
+	authSub.Signature = sign
+	authSub.Timestamp = signTime
+	authSub.MessageType = "subscribe"
+	return b.WebsocketConn.SendMessage(authSub)
```

Comment:
This prevents the websocket from being able to track what channels it is subscribed to.

Please replace with `b.Websocket.SubscribeToChannels(subscriptions)` and have the `Subscribe` function handle the authsending where the channel is an auth channel